_posts/2019-05-23-quarkus-and-angular-ui-development-mode.adoc

```diff
@@ -0,0 +1,391 @@
+. Use the Angular proxy to proxy REST calls to the back-end, while handling the 'local' web application routes in the browser. Other modern web application frameworks should have something similar.
+. We need to use the proxy url when we have a servlet on the back-end which needs to redirect back to the client. Also, when linking to servlets on the back-end we need to update the links on the client.
+
+All the code is hosted on GitHub at https://github.com/kabir/blog-quarkus-ui-development. There is a tag called `step-1`, `step-2` etc. for each of the below steps, and each of the tags contains one commit. You can either follow the steps in this post (I will give the important snippets of code, but also refer back to the individual commits for the full code), or you can clone the GitHub repository and check out the tags to save some typing.
+
+Although this post goes into a lot of detail in setting up our sample application, the actual changes needed to make it behave well are quite https://github.com/kabir/blog-quarkus-ui-development/compare/step-3..step-6[trivial].
+
+
+This post will go through the following steps:
+
+1. Bootstrapping the Quarkus and Angular applications
+2. Add Maven plugins to copy the built web application to the right location for bundling into the Quarkus application
+3. Add some classes containing the main demo code. Everything up to and including this step is really just to have something to illustrate the problems, so if you are impatient you can jump ahead to step 4 where we start adding the things mentioned above. But make sure you run `mvn package -Dui.deps -Dui.dev` before trying to run anything!
+4. Add a servlet filter to let Angular deal with the URLs that are meant for it. This addresses a) from above.
+5. Set up the Angular proxy for the dual development mode. This addresses b) from above.
+6. A quick way to redirect back to the proxy from a back-end servlet, and to change the links to servlets. This addresses c) from above.
+
+Let's get started!
+
+==== Prerequisites
+You need to have https://nodejs.org/en/download/package-manager/[Node], https://yarnpkg.com/en/docs/install[Yarn] and https://cli.angular.io[Angular CLI] installed on your system.
+
+Familiarity with Angular and Quarkus is assumed.
+
+==== Step 1 - Bootstrapping the Quarkus and Angular applications
+
+The diff for this step can be found https://github.com/kabir/blog-quarkus-ui-development/compare/step-0..step-1[here].
+
+First off scaffold the project (substitute the '0.15.0' with the latest and greatest Quarkus release version):
+
+```
+$mvn io.quarkus:quarkus-maven-plugin:0.15.0:create \
+    -DprojectGroupId=org.kabir.quarkus \
+    -DprojectArtifactId=blog-quarkus-ui-development \
+    -DprojectVersion=0.1.0 \
+    -DclassName="org.kabir.quarkus.ui.SampleResource" \
+    -Dextensions=io.quarkus:quarkus-resteasy,io.quarkus:quarkus-resteasy-jsonb
```

Comment:
` -Dextensions=io.quarkus:quarkus-resteasy-jsonb` is sufficient

quarkus-resteasy is already added in template - https://github.com/quarkusio/quarkus/blob/master/devtools/common/src/main/resources/templates/basic-rest/java/pom-template.ftl#L31

Right now user will see warnings like:
```
[WARNING] 'dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique:
   io.quarkus:quarkus-resteasy:jar -> duplicate declaration of version (?) @ line 46, column 17
```